### PR TITLE
Replace "Create New" and "My Projects" buttons with expanded existing Create button

### PR DESCRIPTION
**Pull request by Hamms**
[Spec](https://docs.google.com/document/d/1V0ApTIbg7qWIIq9pDcJ2kdP1rcuxLKi2lmKXkYdIdVA/edit#)
[Jira](https://codedotorg.atlassian.net/browse/FND-82)
Depends on https://github.com/code-dot-org/code-dot-org/pull/27576

- Modify the "Create" dropdown to dynamically include the active project type at the top of the list, when appropriate
- Also restyle it to more closely match the user menu (when hovering, apply a shade rather than a bold)
- Replace the "Create New" and "My Projects" buttons in the header with the new Create button

Peek 2019-03-19 12-03

### CODE REVIEW

File: lib/cdo/create_header.rb [hunk 1]
```diff
@@ -33,11 +49,15 @@ def self.get_create_dropdown_contents(options)
     applab_gamelab = %w(applab gamelab)
 
     entries = options[:limit_project_types] == "true" ?
-      everyone_entries + ["minecraft"] :
+      everyone_entries + ["minecraft_designer"] :
       everyone_entries + applab_gamelab
 
     entries << "dance"
```
Comment (davidsbailey) [inline lib/cdo/create_header.rb:55]: this is a fair bit harder to follow than:
```ruby
entries = options[:limit_project_types] ?
  %w{spritelab artist minecraft_designer dance} :
  %w{spritelab artist applab gamelab dance}
```

**Review by davidsbailey — COMMENTED** [review]
I think some of this code changed while I was reviewing -- please ping again when this is ready for another look.

File: lib/cdo/create_header.rb (hunk 1 above)
Comment (Hamms) [reply lib/cdo/create_header.rb:55]: oh, good catch. I went with the smallest delta from how it currently is for my first pass, then forgot to go back and clean it up. Thanks!

**Comment by Hamms** [comment]
Oh, sorry about that. I realized this branch included the commits from the dependent PR https://github.com/code-dot-org/code-dot-org/pull/27576 in the diff, so I changed the base branch of this PR to simplify things. It should be good to review now

**Review by davidsbailey — APPROVED** [review]
no worries, looks good!

**Review by breville — APPROVED** [review]
Great styling improvements.  I've always wished the dropdown's right edge aligned with the button's right edge, so that all our dropdowns are consistent like that, but not crucial here.

**Pull request merged**